tests/moduletool/test_module_tool_basics.py

```diff
@@ -332,3 +334,47 @@ async def test_module_requires_legacy_invalid(inmanta_module):
     )
     with pytest.raises(InvalidMetadata, match="Invalid legacy requires"):
         module.Module(None, inmanta_module.get_root_dir_of_module())
+
+
+def test_project_repo_type_module_v2(modules_dir, modules_repo, caplog):
+    """
+    Tests that repos that are strings and repos that are dict with
+    type 'git' are accepted and that repos with another type
+    will raise a warning. (issue #3565)
+    """
+    makeproject(modules_repo, "project", [], [])
```

Comment:
If I understand correctly this creates a directory in a shared directory, is that right? I really don't like how these tests are set up but that's out of our hands. I do think though that it would be a good idea to use a slightly more unique name to prevent clashes with future tests. Perhaps something with the name of this test, or even a uuid?